src/sbvr-api/permissions.ts

```diff
@@ -387,12 +392,60 @@ const $checkPermissions = (
 	});
 };
 
-const constrainedPermissionError = new PermissionError();
-const generateConstrainedAbstractSql = (
+const convertToLambda = (filter: AnyObject, identifier: string) => {
+	// We need to inject all occurences of properties for the new lambda function
+	// for example if there is an `or` operator, where different properties of
+	// the target resource are used, we need to change all occurences of these.
+	// We do this in a recursive way to also cover all nested cases like or [ and, and ]
+	const replaceObject = (object: AnyObject) => {
+		if (typeof object === 'string') {
+			return;
+		}
+		if (Array.isArray(object)) {
+			object.forEach(element => {
+				replaceObject(element);
+			});
+		}
+
+		if (object.hasOwnProperty('name')) {
```

Comment:
It doesn't make sense to check `object.hasOwnProperty('name')` if `object` is actually an array
```suggestion
		if (Array.isArray(object)) {
			object.forEach(element => {
				replaceObject(element);
			});
		} else if (object.hasOwnProperty('name')) {
```